server/api.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+type API struct {
+	plugin *Plugin
+	router *mux.Router
+}
+
+func (a *API) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	a.router.ServeHTTP(w, r)
+}
+
+func (a *API) handlerRemoveAttachments(w http.ResponseWriter, r *http.Request) {
+	// Get post_id from the query parameters
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	postID := r.FormValue("post_id")
+	if !model.IsValidId(postID) {
+		http.Error(w, "Invalid post ID", http.StatusBadRequest)
+		return
+	}
+
+	// Check if the user is the post author or a system admin
+	userID := r.Header.Get("Mattermost-User-ID")
+	if !model.IsValidId(userID) {
+		http.Error(w, "Invalid user ID", http.StatusBadRequest)
+		return
+	}
+
+	post, appErr := a.plugin.API.GetPost(postID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	if errReason := a.plugin.userHasRemovePermissionsToPost(userID, post.ChannelId, postID); errReason != "" {
+		http.Error(w, errReason, http.StatusForbidden)
+		return
+	}
+
+	// Otherwise, update the post without attachments and then soft-delete the attachments from the channel
+	originalFileIDs := post.FileIds
+
+	post.FileIds = []string{}
+	newPost, appErr := a.plugin.API.UpdatePost(post)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	// Soft-delete the attachments from channel
+	for _, fileID := range originalFileIDs {
+		if err := a.plugin.SQLStore.DetachAttachmentFromChannel(fileID); err != nil {
+			a.plugin.API.LogError("error detaching attachment from channel", "err", err)
+			http.Error(w, "Internal server error, check logs", http.StatusInternalServerError)
+			return
+		}
+	}
```

Comment:
Just checking that we wouldn't want to proceed on error here with just logging it. 0/5 on that being better, but just wanted to check. When doing Wrangler stuff I sometimes opted to try to finish a given task once a "breaking change" was made and just log errors. Once again, just checking if that is something we thought about for unhappy paths.

Reply:
@fmartingr I'll let you decide if you want do this via a new PR.  A new PR could make best efforts to delete each attachment, continuing on error and logging.  The attaching IDs to original post would then work off a new slice built in this loop.